src/binary/ContractionOptimizer.h

```diff
@@ -32,6 +32,12 @@ class einsum_ir::binary::ContractionOptimizer {
     //! targeted number of tasks
     int64_t m_target_parallel = 1;
 
+    //! number of bytes for scalar data types in output tensor
+    int64_t m_num_bytes_scalar_out = 4;
+
+    //! size of L2 cache in bytes
+    int64_t m_l2_cache_size = 1024 * 1024; // 1MB L2 cache size
+
```

Comment:
Init invalid and set defaults in the constructor for consistency with the remaining code. Also for currently uninitialized values, e.g., ``m_size_all_m`` and ``m_size_all_n``.